### DISCUSSION TITLE
Meeting notes

### DISCUSSION THREAD

**Discussion by GForb** [pmsims-package/pmsims#7] [General]
In this discussion we can post summaries from our meetings. Each meeting can be posted as a new comment.

limit discussion in this thread to what we did or did not cover in the meeting. If we want to discuss topics from the meeting in more detail they should be given their own discussion thread.

**Comment by GForb** [comment]
# Meeting 22/02/2023

My recollection of yesterdays discussion is:

## How this fits in with Felix's other packages
It makes sense to have this as a stand alone package and paper, possibly for [Jstatsoft](https://www.jstatsoft.org/index). One of Felxi's other projects could be implemented as wrapper for this.

## Surrogate modelling
- There are a few different options. One is to use the mlrMBO package, other's include approaches in Felix's other package. 
- An advantage of gausian process approaches is they can give you the optimal next point to sample
- I sugested quantile regression but have never played with them so don't know how they could perform.

## Tuning
We discussed whether tuning a single parameter in the data generating model is enough. I cannot remember everything that was said but there was a view that multi parameter tuning would be helpful for users.

## Code reorganization

GF has reorganised Deborah's code to separate out the different components:
- tune_output takes a data generating mdoel and a target performance and optimises a single parameter in the data generating mdoel
- calculate_sample_size does the simulation part - this is where any surrogate modelling would reside
- The example script contains the fucntions for the data generating model, the analysis model and the performance metric as part of a full example comparing the sample size needed for vanilla logistic vs lassoo. 

## What the user does vs what the user gets
As well as allowing the user to specify their own functions we would want to supply the user with some default data generating models and performance metric functions.


Add in replies to this anything else.

**Comment by GForb** [comment]
# Agenda for meeting 14/03/2022

What do we want to speak about?

Some ideas from me:

1. How to organise  ourselves to make the most of each of our skills, time and energy for this project
2. Surrogate modelling - I should have an example of a quantile gam approach I could show and tell
3. Pick up on any/all of the discussion threads on this page

**Reply by ewancarr** [reply]
In addition to (1), we should probably discuss timelines. Because without a deadline, it's hard to prioritise.

**Reply by flxzimmer** [reply]
To (2) I could also contribute some code (the simulation_mlpwr.html in my branch). It accounts for uncertainty (via bootstrapping) and searches for quantiles of the performance. It uses my package but for integration to this package a lightweight version of this using mlrMBO could be doable.
